src/main/java/carpet/script/value/FormattedTextValue.java

```diff
@@ -27,7 +27,7 @@ public boolean getBoolean() {
     @Override
     public Value clone()
     {
-        return new FormattedTextValue(text);
+        return new FormattedTextValue(text.shallowCopy());
```

Comment:
I used clone only in the cases of assigning variables, and since text values are immutable here (and you kept that), this is not needed.